src/emulator/pubsubEmulator.ts

```diff
@@ -66,11 +74,18 @@ export class PubsubEmulator implements EmulatorInstance {
     return Emulators.PUBSUB;
   }
 
-  async addTrigger(topicName: string, trigger: string) {
-    this.logger.logLabeled("DEBUG", "pubsub", `addTrigger(${topicName}, ${trigger})`);
+  async addTrigger(topicName: string, triggerKey: string, signatureType: SignatureType) {
+    this.logger.logLabeled(
+      "DEBUG",
+      "pubsub",
+      `addTrigger(${topicName}, ${triggerKey}, ${signatureType})`
+    );
 
-    const topicTriggers = this.triggers.get(topicName) || new Set();
-    if (topicTriggers.has(topicName) && this.subscriptions.has(topicName)) {
+    const triggers = this.triggersForTopic.get(topicName) || [];
+    if (
+      triggers.some((t) => t.triggerKey === triggerKey) &&
+      this.subscriptionForTopic.has(topicName)
+    ) {
       this.logger.logLabeled("DEBUG", "pubsub", "Trigger already exists");
```

Comment:
Couldn't two functions listen to the same topic?

Reply:
It was a bit confusing, but I learned that in the PubSub emulator, there is a single subscription per pubsub topic, and the events sent on that subscription is routed to all triggers listening on the topic.

Reply:
Yes that should work?